src/main/scala/org/ergoplatform/api/routes/InfoRoute.scala

```diff
@@ -19,14 +19,18 @@ import scala.concurrent.Future
 case class InfoRoute(readersHolder: ActorRef,
                      miner: ActorRef,
                      peerManager: ActorRef,
-                     digest: Boolean,
-                     override val settings: RESTApiSettings, nodeId: Array[Byte])
+                     ergoSettings: ErgoSettings,
+                     nodeId: Array[Byte])
                     (implicit val context: ActorRefFactory) extends ErgoBaseApiRoute {
   override val route = info
 
+  override val settings: RESTApiSettings = ergoSettings.scorexSettings.restApi
+
   private def getConnectedPeers: Future[Int] = (peerManager ? PeerManager.GetConnectedPeers).mapTo[Seq[Handshake]].map(_.size)
 
-  private def getStateType: String = if (digest) "digest" else "utxo"
+  private def getStateType: String = if (ergoSettings.nodeSettings.ADState) "digest" else "utxo"
```

Comment:
It's always better to encode (if possible) possible states in types.

Reply:
Created issue https://github.com/ergoplatform/ergo/issues/174